samples/logging/prometheus-grafana-opentelemetry/sample.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Use OpenTelemetry with Prometheus and Grafana
+summary: Use OpenTelemetry with Prometheus and Grafana to capture and visualize NServiceBus metrics.
+component: Core
+isLearningPath: true
+reviewed: 2022-06-24
+previewImage: grafana.png
+---
+
+
+## Introduction
+
+[Prometheus](https://prometheus.io) is a monitoring solution for storing time series data like metrics. [Grafana](https://grafana.com) allows to visualize the data stored in Prometheus (and other sources). This sample demonstrates how to capture NServiceBus OpenTelemetry metrics, store them in Prometheus and visualize these metrics using a Grafana dashboard.
+
+
+![Grafana NServiceBus fetched, processed and errored messages](grafana.png)
+
+## Prerequisites
+
+To run this sample, Prometheus and Grafana are required. This sample uses docker and an accompanied `docker-compose.yml` file to run the stack.
+
+## Code overview
+
+The sample simulates messages load with a random 10% failure rate using the `LoadSimulator` class:
+
+snippet: prometheus-load-simulator
+
+## Reporting metric values
+NServiceBus version 8.x uses OpenTelemetry standard to report metrics. The metrics are disabled by default, and can be enabled:
+
+snippet: enable-opentelemetry-metrics
+
+There are three metrics reported as a Counter:
+
+ * Number of fetched messages 
+ * Number of failed messages
+ * Number of successfully processed messages
+
+Each reported metric is tagged with the following additional information:
+
+ * `messaging_discriminator` a uniquely addressable address for the endpoint (discriminator when scaling out)
+ * `messaging_endpoint` endpoint name
+ * `messaging_queue` the queue name of the endpoint
+ * `messaging_type` the .NET type information for the message being processed
+ * `messaging_failure_type` the exception type name (if applicable)
+
+## Exporting metrics
+
+The metrics are gathered using OpenTelemetry standard on the endpoint and need to be reported and collected by an external service. A Prometheus exporter can expose this data to the Prometheus service, hosted as a docker service. This exporter is available via a NuGet package `OpenTelemetry.Exporter.Prometheus`. The sample exposes an HTTP endpoint that enables Prometheus to scrape data reported by the OpenTelemetry metrics. In the sample the service that exposes the data to scrape is hosted on `http://localhost:9185/metrics`. To enable the Prometheus exporter, the following should be executed:
```

Comment:
We might benefit from having a diagram here showing how these components interact

Reply:
Added a diagram. See what you think.